extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts

```diff
@@ -276,10 +280,11 @@ export class ChePluginManager {
    */
   private async getAllPlugins(filter: string): Promise<ChePlugin[]> {
     // get list of all plugins
-    const rawPlugins = await this.chePluginService.getPlugins(filter);
+    const rawPlugins = await this.chePluginService.getPlugins();
+    const filteredPlugins = PluginFilter.filterPlugins(rawPlugins, filter);
 
     // group the plugins
-    const grouppedPlugins = this.groupPlugins(rawPlugins);
+    const grouppedPlugins = this.groupPlugins(filteredPlugins);
```

Comment:
I would not have mixed code refactoring and new feature, it's almost impossible to review what is really new

Reply:
Sometimes it needs to refactor few lines of code (if we speak about this method) to avoid writing something, that will be removed in the next PRs.
In this case, the filtering performs on the client side. It's just to avoid duplicating the code for both implementations.